### ISSUE TITLE
Bug with macro IF statement, system expects CBA to hold number but it holds identifier to a number

### ISSUE THREAD

**Issue by nibblebits** [nibblebits/DragonCompiler#43]
```
#define ABC 50
#define CBA ABC

#if CBA == 50
#warning "valid"
#endif
```

**Comment by nibblebits** [comment]
Above code should work with no problems but it doesnt.

**Comment by nibblebits** [comment]
Fixed: e5ddb02bbb31380f7dc2a5519efaf2a426f04bc4
